tests/test_optimizer/test_dist_adafactor.py

```diff
@@ -0,0 +1,626 @@
+    base_optim = Adafactor(base_param_group)
+    dist_optim = DistributedAdaFactor(tp_param_group)
+
+    # Setup distributed optimizer
+    if zero_size > 1:
+        base_optim = LowLevelZeroOptimizer(
+            base_optim,
+            overlap_communication=True,
+            initial_scale=128,
+            partition_grad=True,
+            dp_process_group=dp_group,
+            verbose=True,
+        )
+
+        dist_optim = LowLevelZeroOptimizer(
+            dist_optim,
+            overlap_communication=True,
+            initial_scale=128,
+            partition_grad=True,
+            dp_process_group=dp_group,
+            verbose=True,
+        )
+        shard_to_param = dist_optim._param_store.master_to_working_param  # {id(): param tensor} but flattened
+        dist_optim.optim.setup_distributed(
+            tensor_parallel_group=tp_group,
+            data_parallel_group=dp_group,
+            shard_to_param=shard_to_param,
+            use_zero=use_zero,
+        )
+    else:
+        shard_to_param = set_master_param_to_shard_param(tp_param_group)
+        dist_optim.setup_distributed(
+            tensor_parallel_group=tp_group,
+            data_parallel_group=dp_group,
+            shard_to_param=shard_to_param,
+            use_zero=use_zero,
+        )
+
+    # ==============================
+    # Correctness Verify
+    # ==============================
+    x = torch.randn(HEIGHT, WIDTH, device=local_rank)
+
+    out = base_model(x)
+    out_tp = tp_model(x)
+
+    if zero_size > 1:
+        dist_optim.backward(out_tp.sum())
+        base_optim.backward(out.sum())
+    else:
+        out_tp.sum().backward()
+        out.sum().backward()
+
+    base_optim.step()
+    dist_optim.step()
+
+    base_optim.zero_grad()
+    dist_optim.zero_grad()
+
+    for p, tp_p in zip(base_param_group, tp_param_group):
+        param_is_distributed = is_distributed_tensor(tp_p)
+        if param_is_distributed:
+            shard_spec = get_sharding_spec(tp_p)
+            if len(shard_spec.sharding_sequence) >= 2:
+                # Col Parallel
+                if shard_spec.sharding_sequence[0] == "R":
+                    tp_p = _gather(input_=tp_p, dim=-1, process_group=tp_group)  # gather
+                # ROW Parallel
+                if shard_spec.sharding_sequence[-1] == "R":
+                    tp_p = _gather(input_=tp_p, dim=0, process_group=tp_group)  # gather
+            else:
+                # TP bias
+                tp_p = _gather(input_=tp_p, dim=-1, process_group=tp_group)  # gather
+        else:
+            # No TP bias
+            pass
+        correctness_verify(p.data, tp_p.data, dtype)
+    clear_layout_converter()
+    Randomizer.reset_index()
+    torch.cuda.empty_cache()
+    print(f"Zero Test Pass")
+
+
+@parameterize("dtype", [torch.float16])
+@parameterize("tp_zero_size", [(1, 4)])
+def exam_dist_adafactor_booster(dtype: torch.dtype, tp_zero_size: tuple[int, int]):
```

Comment:
This is not recommended usage. TP should be used with hybriad parallel plugin, rather than low level zero plugin. And users should not use LowLevelZeroOptimizer directly when booster is used.

Reply:
> This is not recommended usage. TP should be used with hybriad parallel plugin, rather than low level zero plugin. And users should not use LowLevelZeroOptimizer directly when booster is used.


Sure.  TP shouldn't appear with low level zero plugin, and in new push:https://github.com/hpcaitech/ColossalAI/compare/4a28efedf2839a79206f48b5edb4348c083775b0..0a7f68269077347655d5c2bc284320bac29d1116 , I add test for "lowlevelzeroPlugin + Bert model zoo"  and "HybridPlugin(with diff dp,tp combinations) + Bert model zoo". 
Should I removed exam_dist_adafactor_zero and exam_dist_adafactor_booster.